### PR TITLE
batch appends

### PR DESCRIPTION
**Pull request by mvayngrib**
based on #4

### CODE REVIEW

**Comment by mafintosh** [comment]
@mvayngrib thanks for all the PRs. I'll go through them all over the weekend :)

**Comment by mvayngrib** [comment]
@mafintosh great, thanks

File: index.js [hunk 1]
```diff
@@ -23,15 +20,38 @@ module.exports = function(db) {
...
+        batch = batch.slice()
+        feed.batch.length = 0
+
+        db.batch(batch.map(function (item) {
+          return {
+            type: 'put',
+            key: item.key,
+            value: item.value
+          }
+        }), function(err) {
```
Comment (mafintosh) [inline index.js:54]: if the feed fails do we reset the change count?

**Comment by mafintosh** [comment]
@mvayngrib btw, do you want to help maintain this module? i can add you as collaborator if you want :)

File: index.js (hunk 1 above)
Comment (mvayngrib) [inline index.js:54]: good point

**Comment by mvayngrib** [comment]
@mafintosh yes, add me please :)

File: index.js (hunk 1 above)
Comment (mvayngrib) [inline index.js:54]: but if one batch fails and meanwhile the next succeeds? Resetting might make things even weirder. I guess this is where mutefixy kept things simple

Comment (mafintosh) [inline index.js:54]: yea this gets a bit tricky fast. maybe we should add some tests (somehow) for this so we can make we hit all the edge-cases

**Comment by mafintosh** [comment]
@mvayngrib added. whats your npm username?

File: index.js (hunk 1 above)
Comment (mafintosh) [inline index.js:54]: i think the simplest solution is:
1. set a flag that we are running a batch.
2. if flag is set append to a known tmp batch
3. after 1 completes unset flag and if tmp batch is not empty goto 1 and run that batch

basically a mutex batch (should be super fast)

Comment (mvayngrib) [inline index.js:54]: right, the tmp batch we already have, it's `feed.batch`. We just need a lock around the commit so there is never the possibility of commit 2 succeeding while commit1 fails.

the question is what to do when the current batch fails, and there's also a tmp batch in the balance

**Comment by mvayngrib** [comment]
@mafintosh npm username: tenaciousmv  (also, see line note above re: batch locks)
